[material-ui][Slider] Set CSS `writing-mode` and update vertical slider docs (#44537)
mui/material-ui: docs/data/material/components/slider/slider.md

## Vertical sliders

Set the `orientation` prop to `"vertical"` to create vertical sliders. The thumb will track vertical movement instead of horizontal movement.

{{"demo": "VerticalSlider.js"}}

:::warning

> Chrome versions below 124 implement `aria-orientation` incorrectly for vertical sliders and exposes them as `'horizontal'` in the accessibility tree. ([Chromium issue #40736841](https://issues.chromium.org/issues/40736841))

Reviewer: ```suggestion
Chrome versions below 124 implement `aria-orientation` incorrectly for vertical sliders and expose them as `'horizontal'` in the accessibility tree. ([Chromium issue #40736841](https://issues.chromium.org/issues/40736841))
```

Reviewer: Ping

> Chrome versions below 124 implements `aria-orientation` incorrectly for vertical sliders and exposes them as `'horizontal'` in the accessibility tree. ([Chromium issue #40736841](https://issues.chromium.org/issues/40736841))

Reviewer: ```suggestion
Chrome versions below 124 implement `aria-orientation` incorrectly for vertical sliders and expose them as `'horizontal'` in the accessibility tree. ([Chromium issue #40736841](https://issues.chromium.org/issues/40736841))
```

> Chrome versions below 124 implements `aria-orientation` incorrrectly for vertical sliders and exposes them as `'horizontal'` in the accessibility tree. ([Chromium issue #40736841](https://issues.chromium.org/issues/40736841))

Reviewer: ```suggestion
Chrome versions below 124 implements `aria-orientation` incorrectly for vertical sliders and exposes them as `'horizontal'` in the accessibility tree. ([Chromium issue #40736841](https://issues.chromium.org/issues/40736841))
```

> Chrome versions below 124 implements `aria-orientation` incorrrectly for vertical sliders and exposes them as `'horizontal'` in the accessibility tree. ([Chromium issue #40736841](https://issues.chromium.org/issues/40736841))

Reviewer: Do you know why we actively support Chrome 109? Why the jump from 109 to 119 in https://github.com/mui/material-ui/blob/v6.1.8/.browserslistrc#L28-L29? I wonder if this warning makes sense now that Chrome is already on v131.

The `-webkit-appearance: slider-vertical` CSS property can be used to correct this for these older versions, with the trade-off of causing a console warning in newer Chrome versions:

```css

> .MuiSlider-thumb > input {

Reviewer: Would `.MuiSlider-thumb input` be a better selector? e.g. if the internal DOM structure changes and `input` is not a direct child in the future, this would break. And since afaik there can only be one `input` it's safe to assume that it will always target the input.

  -webkit-appearance: slider-vertical;
}
```

> For Chrome 124 and newer, the slider includes the CSS [`writing-mode`](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_writing_modes/Vertical_controls#range_sliders_meters_and_progress_bars) property that fixes this bug.

Reviewer 2: Users don't have to add this themselves, right? I would remove/rewrite this, otherwise I think people might think they need to provide it themselves.

Author: Updated!
